File: dexp.c
#include <stdio.h>
#include <math.h>

void dexp(double omega[3][1], double u[3][1], double du[3][1]);	    //protoype



void dexp(double omega[3][1], double u[3][1], double du[3][1]){

double norm_u = 0;
double skew_u[3][3] = {0};
double SO3_matrix[3][3] = {0};
norm_u = sqrt(pow(u[0][0],2)+pow(u[1][0],2)+pow(u[2][0],2));


skew(u, skew_u);		
skew(omega, SO3_matrix);	
SO3_to_R3(SO3_matrix, du);	

}

File: python-implementation/dexp.py
import numpy as np

from skew import skew
from SO3_to_R3 import SO3_to_R3
from lie_bracket import lie_bracket


def dexp(omega,u):
    norm_u = np.linalg.norm(u)
    skew_u = skew(u)
    du = SO3_to_R3(skew(omega))
    

    return du


